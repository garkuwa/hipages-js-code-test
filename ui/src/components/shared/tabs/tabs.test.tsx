import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import Tabs from './tabs';

describe('Tabs', () => {
    const leftClick = { button: 0 };

    it('should render Tabs with three tabs', () => {
        const tabs = renderer.create(
            <Tabs>
                <Tabs.Tab name="One">One</Tabs.Tab>
                <Tabs.Tab name="Two">Two</Tabs.Tab>
                <Tabs.Tab name="Three">Three</Tabs.Tab>
            </Tabs>,
        );

        expect(tabs).toMatchSnapshot();
    });

    it('should call the tab change handler on a tab change', () => {
        const tabChange = jest.fn();
        const { getByTestId } = render(
            <Tabs onTabChange={tabChange}>
                <Tabs.Tab name="One">One</Tabs.Tab>
                <Tabs.Tab name="Two">Two</Tabs.Tab>
                <Tabs.Tab name="Three">Three</Tabs.Tab>
            </Tabs>,
        );
        fireEvent.click(getByTestId('One'), leftClick);

        expect(tabChange).toBeCalledTimes(1);
    });

    it('should set an active tab', () => {
        const { getByTestId } = render(
            <Tabs activeTabIndex={2}>
                <Tabs.Tab name="One">One</Tabs.Tab>
                <Tabs.Tab name="Two">Two</Tabs.Tab>
                <Tabs.Tab name="Three">Three</Tabs.Tab>
            </Tabs>,
        );

        expect(getByTestId('Two')).toBeInTheDocument();
    });
});
