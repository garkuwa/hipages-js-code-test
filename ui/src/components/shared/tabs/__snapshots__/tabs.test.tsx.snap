// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render Tabs with three tabs 1`] = `
<div
  className="sc-bdnxRM dsaolF"
>
  <div
    className="sc-gtsrHT hnHYbc"
  >
    <div
      className="sc-dlnjwi jgsURa"
      data-testid="One"
      onClick={[Function]}
    >
      One
    </div>
    <div
      className="sc-dlnjwi hHTMSR"
      data-testid="Two"
      onClick={[Function]}
    >
      Two
    </div>
    <div
      className="sc-dlnjwi hHTMSR"
      data-testid="Three"
      onClick={[Function]}
    >
      Three
    </div>
  </div>
  <div
    className="sc-hKFxyN bpCJQI"
  >
    One
  </div>
</div>
`;
